pkg/flow/flow_mem_driver.go

```diff
@@ -809,6 +809,20 @@ func (fc *FlowCollector) updateRecord(record interface{}) error {
 				}
 			} else {
 				if link.EndTime > 0 {
+					// incoming link being deleted so remove matching outgoing link
+					if current.Direction != nil && *current.Direction == Incoming {
+						// name of peer router is part of link name
```

Comment:
Thank you @lynnemorrison  - I like this approach.

Mind testing this out with a podman site as well first, though? I'm not confident that the way we are correlating a router name to a link parent ID is something the router holds as a stable part of its API. That is, there may be scenarios where there isn't exactly 5 dashes in the router name (podman, gateways, kube sites with a dash in the namespace), and it could be that the router's ROUTER record may not always use `<prefix>:0` as its id.

Reply:
@c-kruse is there a better way to get the router id from a link name?  I looked through the code to see if there were any macos/functions but couldn't find any.

Reply:
@lynnemorrison I'm not positive there's an excellent example going in this direction. That `graph` function and the linkResponseHandler get there in slightly different ways. My understanding is that you're pretty close, but maybe missing a condition or two. Links can get pretty exciting with corner cases.

I'd recommend looking at using normalizeRouterName - with that you should be able to consistently get the peer router name of a link. That's what I've been promised is the most concrete way to complete the link-to-peer-router relation.

So starting with a subject incoming link, I think you'd want to find that subject link's router and note the router name. Then iterate peer links: Links where the direction is outgoing and where the link's normalized name matches our subject link's router name. Then finally check that that peer link's router's name matches the subject link's normalized name.